src/page-modules/assistant/layout.tsx

```diff
@@ -0,0 +1,222 @@
+import type { GeocoderFeature } from '@atb/page-modules/departures';
+import { PageText, useTranslation } from '@atb/translations';
+import { FormEventHandler, PropsWithChildren, useState } from 'react';
+import Search from '@atb/components/search';
+import { Button } from '@atb/components/button';
+import style from './assistant.module.css';
+import { useRouter } from 'next/router';
+import DepartureDateSelector, {
+  DepartureDate,
+  DepartureDateState,
+} from '@atb/components/departure-date-selector';
+import TransportModeFilter from '@atb/components/transport-mode-filter';
+import { Typo } from '@atb/components/typography';
+import {
+  filterToQueryString,
+  getInitialTransportModeFilter,
+} from '@atb/components/transport-mode-filter/utils';
+import { TransportModeFilterOption } from '@atb/components/transport-mode-filter/types';
+import { MonoIcon } from '@atb/components/icon';
+import { FocusScope } from '@react-aria/focus';
+import { featuresToFromToQuery } from './utils';
+
+export type AssistantLayoutProps = PropsWithChildren<{
+  initialFromFeature?: GeocoderFeature;
+  initialToFeature?: GeocoderFeature;
+  initialTransportModesFilter?: TransportModeFilterOption[] | null;
+}>;
+
+function AssistantLayout({
+  children,
+  initialFromFeature,
+  initialToFeature,
+  initialTransportModesFilter,
+}: AssistantLayoutProps) {
+  const { t } = useTranslation();
+  const router = useRouter();
+
+  const [showAlternatives, setShowAlternatives] = useState(false);
+  const [selectedFromFeature, setSelectedFromFeature] = useState<
+    GeocoderFeature | undefined
+  >(initialFromFeature);
+  const [selectedToFeature, setSelectedToFeature] = useState<
+    GeocoderFeature | undefined
+  >(initialToFeature);
+  const [departureDate, setDepartureDate] = useState<DepartureDate>({
+    type: DepartureDateState.Now,
+  });
+  const [transportModeFilter, setTransportModeFilter] = useState(
+    getInitialTransportModeFilter(initialTransportModesFilter),
+  );
+
+  const onSwap = () => {
+    let query = {};
+    const filter = filterToQueryString(transportModeFilter);
+
+    if (filter) {
+      query = {
+        filter,
+      };
+    }
+
+    const arriveBy = departureDate.type === DepartureDateState.Arrival;
+    if (arriveBy) {
+      query = {
+        ...query,
+        arriveBy,
+      };
+    }
+
+    const when = departureDate.type !== DepartureDateState.Now;
+    if (when) {
+      query = {
+        ...query,
+        when: departureDate.dateTime,
```

Comment:
Is both `when` and `arriveBy` needed if e.g. `departureDate.type === DepartureDateState.Arrival`? Instead of `arriveBy` _or_ `departBy` that is.

Reply:
Ah, see that `arriveBy` is a boolean now. But the question sort of still stands.

Reply:
We could skip the `arriveBy` when it's false. The date can also be set server side, so if it is not sent we use "now". I'm not sure if leaving out those parameters affects anything else or if it improves the code though 🤔

Reply:
How does the querying deal with "depart by"?

Reply:
That would be `arriveBy = false` and `when` the date you would like to depart by. Similar to "now", but with different date

Reply:
Alright, I've updated this to use the `arriveBy` and `departBy` values.

Reply:
I agree! Good suggestion 🎉